api/logs/logSender.go

```diff
@@ -103,7 +105,8 @@ func (ls *logSender) waitForPatternMessage() error {
 func (ls *logSender) monitorConnection() {
 	for {
 		mt, _, err := ls.conn.ReadMessage()
-		if mt == websocket.CloseMessage {
+		ls.log.Info("message type", "value", mt)
+		if mt == websocket.CloseMessage || mt == disconnectMessage {
 			_ = ls.writer.Close()
```

Comment:
As stated, I think it is better to close the writer if any error is received. Maybe add a defer func on monitorConnection?

Reply:
Done.